### PR TITLE
Run tests inside librecores CI docker image

### PR DESCRIPTION
**Pull request by Nancy-Chauhan**
**Goal** : This will create shared tooling of Librecores CI image with or1k_marocchino repository.
It is also in reference to the issue https://github.com/openrisc/mor1kx/issues/81.

### CODE REVIEW

**Comment by Nancy-Chauhan** [comment]
@oleg-nenashev @stffrdhrn I have created a PR in reference to our conversation on https://github.com/openrisc/mor1kx/pull/82 for doing the similar changes in or1k_marocchino core.

**Review by stffrdhrn — COMMENTED** [review]
You created a few commits that just have text `resolving errors`.  That is not very meaningful.  Some points.

1. Commit messages should explain what you are doing and why.
2. It would be good to explain what error each issue is fixing, because in the future this change might cause an issue and it would be good to know why it was done.
3. Learn about commit squashing, usually commits like typo fixes and test fixes can be squashed, so do that.

**Comment by stffrdhrn** [comment]
I think this looks good.  Can you add some documentation in the README.md under the testing section that links to the librecores-ci image and maybe some details about what it is.

**Comment by Nancy-Chauhan** [comment]
@stffrdhrn Yes I am sorry for the commit messages, I will squash it. I will also add some documentation in the README.md. Thank you for your feedback

File: README.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ details please see the [LICENSE](./LICENSE) file or http://juliusbaxter.net/ohdl
 
 A CPU core cannot be trusted without a full set of verification testing.  The
 `or1k_marocchino` is constantly verified for correctness with the or1k Continuous
-Integration (CI) suite running in [travis ci](travis-ci.org).  This currently covers:
+Integration (CI) suite running in [travis ci](https://travis-ci.org/openrisc/or1k_marocchino/).  This currently covers:
```
Comment (oleg-nenashev) [inline README.md]: Would be better to add a badge to the repo

File: .travis/Dockerfile [hunk 2]
```diff
@@ -12,7 +12,7 @@
 #  'RUN .travis/install-verilator.sh' to this dockerfile to have an image
 #  with the full environment already installed.
 #
-FROM ubuntu:trusty
+FROM librecores/librecores-ci:latest
 RUN apt-get update && apt-get install -y \
```
Comment (oleg-nenashev) [inline .travis/Dockerfile]: Many tools below are bundled into the image (for FuseSoC). I suggest cleaning it up.
Also, is this file needed at all after switching to the new base image?

Comment (Nancy-Chauhan) [reply .travis/Dockerfile]: Yes oleg , this file can be removed , though it works on local machine , I will send a commit removing it

File: README.md (hunk 1 above)
Comment (Nancy-Chauhan) [reply README.md]: Badge has already been used below (before my changes) and is currently there too , should  I remove this line ?

File: README.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ details please see the [LICENSE](./LICENSE) file or http://juliusbaxter.net/ohdl
 
 A CPU core cannot be trusted without a full set of verification testing.  The
 `or1k_marocchino` is constantly verified for correctness with the or1k Continuous
-Integration (CI) suite running in [travis ci](travis-ci.org).  This currently covers:
```
Comment (stffrdhrn) [inline README.md:44]: Why remove this link?

File: README.md [hunk 4]
```diff
@@ -51,6 +51,8 @@ Integration (CI) suite running in [travis ci](travis-ci.org).  This currently co
 
    Status: [![Build Status](https://travis-ci.org/openrisc/or1k_marocchino.svg?branch=master)](https://travis-ci.org/openrisc/or1k_marocchino)
 
+[Librecores-CI](https://github.com/librecores/docker-images/tree/master/librecores-ci) docker image is used for testing and continous integration. Librecores CI image includes many common EDA tools such as Icarus Verilog, Verilator, Yosys and also target the FuseSoC use cases. In the travis flow `or1k-test` runs inside a Librecores-CI docker container.
```
Comment (stffrdhrn) [inline README.md]: Maybe you should just look to.
https://github.com/librecores/docker-images

Can you split this line up a bit. Try to limit a line to about 80 characters.  It sould be consistent with other blocks in the readme.

Maybe above mention "... suite running a librecores-ci docker container in Travis ci...."

**Review by stffrdhrn — COMMENTED** [review]
Thanks for the changes. It looks good a few comments.

File: README.md (hunk 3 above)
Comment (Nancy-Chauhan) [reply README.md:44]: Because it is redirected to this link: https://github.com/openrisc/or1k_marocchino/blob/master/travis-ci.org . I will just correct it

File: README.md (hunk 4 above)
Comment (Nancy-Chauhan) [reply README.md]: Did the changes. Please Let me know if it is okay. Thank you

File: README.md (hunk 3 above)
Comment (stffrdhrn) [reply README.md:44]: I see. Thanks for fixing

**Comment by stffrdhrn** [comment]
Thanks I merged this
